### PR TITLE
Msdk 831 improve sonar issues

### PR DESCRIPTION
**Pull request by polymath-eric**
This resolves the Sonar Cloud issues that are not already addressed in alpha

### CODE REVIEW

File: src/api/procedures/modifyCorporateActionsAgent.ts [hunk 1]
```diff
@@ -26,8 +26,6 @@ export type Params = { ticker: string } & ModifyCorporateActionsAgentParams;
 
 /**
  * @hidden
- *
- * @deprecated in favor of `inviteAgent`
```
Comment (polymath-eric) [inline src/api/procedures/modifyCorporateActionsAgent.ts:30]: I figured these are "internal" procedures so we don't need the deprecated comment on them.

File: src/base/__tests__/Context.ts [hunk 2]
```diff
@@ -56,7 +56,6 @@ jest.mock(
   require('~/testUtils/mocks/entities').mockSubsidyModule('~/api/entities/Subsidy')
 );
 
-// TODO: refactor tests (too much repeated code)
```
Comment (polymath-eric) [inline src/base/__tests__/Context.ts:59]: If anything this should be an issue, but I think they are OK even if a bit verbose.

File: src/utils/internal.ts [hunk 3]
```diff
@@ -214,7 +214,6 @@ export function unwrapValues<T extends unknown[]>(values: MapMaybePostTransactio
  */
 type EventData<Event> = Event extends AugmentedEvent<'promise', infer Data> ? Data : never;
 
-// TODO @monitz87: use event enum instead of string when it exists
```
Comment (polymath-eric) [inline src/utils/internal.ts:217]: Looks like its not using a string anymore.

File: src/api/entities/Asset/__tests__/Settlements.ts [hunk 4]
```diff
@@ -236,17 +236,17 @@ describe('Settlements class', () => {
     });
 
     beforeEach(() => {
-      fromPortfolio = entityMockUtils.getDefaultPortfolioInstance({
```
Comment (polymath-eric) [inline src/api/entities/Asset/__tests__/Settlements.ts:239]: It doesn't make sense to me to refactor these into a shared method with those from the `canSettle` case, but by changing the order they shouldn't be the same block as above from the perspective of static analysis.

File: src/Polymesh.ts [hunk 5]
```diff
@@ -347,7 +347,6 @@ export class Polymesh {
     this.context.setPair(signerToString(signer));
   }
 
-  // TODO @monitz87: remove when the dApp team no longer needs it
```
Comment (polymath-eric) [inline src/Polymesh.ts:350]: Seems like this is part of the API now, this comment as been in for a while now.

File: src/testUtils/index.ts [hunk 6]
```diff
@@ -10,3 +13,16 @@ export async function fakePromise(): Promise<void> {
   await new Promise(resolve => setImmediate(() => resolve(null)));
   await new Promise(resolve => setImmediate(() => resolve(null)));
 }
+
+/**
+ * @hidden
+ *
+ * Awaits multiple promises and advances jest time after each one
+ */
+export async function manyFakePromises(): Promise<void> {
```
Comment (prashantasdeveloper) [inline src/testUtils/index.ts:22]:
```suggestion
export async function fakePromises(): Promise<void> {
```
Just a suggestion, but you can leave this as it as well.

**Review by prashantasdeveloper — APPROVED** [review]
Overall LGTM, just left one renaming suggestion.

**Review by monitz87 — APPROVED** [review]

**Review by monitz87 — APPROVED** [review]

**Pull request merged**